packages/core/src/features/genomic/genesFrequencyChartSlice.ts

```diff
@@ -62,125 +50,58 @@ export interface GenesFrequencyChart {
+                  value: ["ssm"],
+                },
+                op: "in",
+              },
+              ...cohortFiltersContent,
+            ],
+            op: "and",
           },
-          ...cohortFiltersContent,
-        ],
-        op: "and",
-      },
-      score: "case.project.project_id",
-    };
-
-    return await graphqlAPI(GeneMutationFrequencyQuery, graphQlVariables);
-  },
-);
-
-export interface GeneFrequencyChartState {
-  readonly frequencies: GenesFrequencyChart;
-  readonly status: DataStatus;
-  readonly error?: string;
-  readonly requestId?: string;
-}
-
-const initialState: GeneFrequencyChartState = {
-  frequencies: { casesTotal: 0, genesTotal: 0, geneCounts: [] },
-  status: "uninitialized",
-};
-
-const slice = createSlice({
-  name: "genomic/geneFrequencyChart",
-  initialState,
-  reducers: {},
-  extraReducers: (builder) => {
-    builder
-      .addCase(fetchGeneFrequencies.fulfilled, (state, action) => {
-        if (state.requestId != action.meta.requestId) return state;
-        const response = action.payload;
-        if (response.errors) {
-          state = castDraft(initialState);
-          state.status = "rejected";
-          state.error = response.errors.filters;
-        }
+          score: "case.project.project_id",
+        };
 
+        return {
+          graphQLFilters: graphQlVariables,
+          graphQLQuery: GeneMutationFrequencyQuery,
+        };
+      },
+      transformResponse: (response) => {
         const data = response.data.geneFrequencyChartViewer.explore;
-        //  Note: change this to the field parameter
-        state.frequencies.casesTotal = data.cases.hits.total;
-        state.frequencies.genesTotal = data.genes.hits.total;
-        state.frequencies.geneCounts = data.genes.hits.edges.map(
-          ({ node }: Record<string, never>) =>
-            (({ gene_id, numCases, symbol }) => ({
-              gene_id,
-              numCases,
-              symbol,
-            }))(node),
-        );
-
-        state.status = "fulfilled";
-        state.error = undefined;
-        return state;
-      })
-      .addCase(fetchGeneFrequencies.pending, (state, action) => {
-        state.status = "pending";
-        state.requestId = action.meta.requestId;
-        return state;
-      })
-      .addCase(fetchGeneFrequencies.rejected, (state, action) => {
-        if (state.requestId != action.meta.requestId) return state;
-        state.status = "rejected";
-        if (action.error) {
-          state.error = action.error.message;
-        }
-        return state;
-      });
-  },
+        return {
+          casesTotal: data.cases.hits.total,
+          genesTotal: data.genes.hits.total,
+          geneCounts: data.genes.hits.edges.map(
+            ({ node }: Record<string, never>) =>
+              (({ gene_id, numCases, symbol }) => ({
+                gene_id,
+                numCases,
+                symbol,
+              }))(node),
+          ),
```

Comment:
Nitpick: 🤯 . Can we make it simpler to understand? (I hope my understanding is correct)
```suggestion
        geneCounts: data.genes.hits.edges.map(({ node }) => ({
             gene_id: node.gene_id,
             numCases: node.numCases,
             symbol: node.symbol,
        })),
```